docker/mini_ota_img/Dockerfile.ota_img

```diff
@@ -0,0 +1,101 @@
+ARG SYS_IMG=ghcr.io/tier4/ota-client/sys_img_for_test:ubuntu_22.04
+ARG UBUNTU_BASE=ubuntu:22.04
+ARG BUSYBOX_VER=busybox:1.37.0
+
+#
+# ------ stage 1: prepare base image ------ #
+#
+
+FROM ${SYS_IMG} AS sys_img
+
+#
+# ------ stage 2: prepare OTA image build environment ------ #
+#
+FROM ${UBUNTU_BASE} AS ota_img_builder
+
+SHELL ["/bin/bash", "-c"]
+ENV DEBIAN_FRONTEND=noninteractive
+
+ENV OTA_METADATA_REPO="https://github.com/tier4/ota-metadata"
+ENV OTA_IMAGE_SERVER_ROOT="/ota-image"
+ENV OTA_IMAGE_DIR="${OTA_IMAGE_SERVER_ROOT}/data"
+ENV CERTS_DIR="/certs"
+ENV SPECIAL_FILE="path;adf.ae?qu.er\y=str#fragファイルement"
+
+COPY --from=sys_img / /${OTA_IMAGE_DIR}
+COPY --chmod=755 ./tests/keys/gen_certs.sh /tmp/certs/
+
+WORKDIR ${OTA_IMAGE_SERVER_ROOT}
+
+RUN set -eux; \
+    apt-get update -qq; \
+    apt-get install -y -qq --no-install-recommends \
+	    gcc \
+        git \
+        libcurl4-openssl-dev \
+        libssl-dev \
+        python3-dev \
+        python3-minimal \
+        python3-pip \
+        python3-venv \
+        wget; \
+    apt-get install -y -qq linux-image-generic; \
+    apt-get clean; \
+    # install uv
+    python3 -m pip install --no-cache-dir -q -U pip; \
+    python3 -m pip install --no-cache-dir uv; \
+    # generate keys and certs for signing
+    mkdir -p "${CERTS_DIR}"; \
+    pushd /tmp/certs; \
+    ./gen_certs.sh; \
+    mv ./* "${CERTS_DIR}"; \
+    popd; \
+    cp "${CERTS_DIR}"/sign.pem sign.pem; \
+    # git clone the ota-metadata repository
+    git clone ${OTA_METADATA_REPO}; \
+    pushd ota-metadata; git checkout 6c8ad47; popd; \
+    # prepare build environment
+    python3 -m venv ota-metadata/.venv; \
+    source ota-metadata/.venv/bin/activate; \
+    python3 -m pip install --no-cache-dir -U pip; \
+    python3 -m pip install --no-cache-dir -q \
+        -r ota-metadata/metadata/ota_metadata/requirements.txt; \
```

Comment:
How about to use the internal operational rule that "We should update this base image when we update dependency". What do you think?

Reply:
Oh, another case when we need to rebuild the test base is, when we want to update the system image we used to build the OTA image.

Reply:
The reason why I suggest is:
For example, in [this PR](https://github.com/tier4/ota-client/pull/537), dependabot suggest to bump up grpcio version. And grpcio deprecates its version at v1.71.0.
- Before this change: we can notice it in the test results in the PR even though we overtook the CHANGELOG.
- After this change: the tests will pass.

So especially when we update dependency, I think it's better to update test base image too. What do you think?

Reply:
Great, It's very clear and stable. Thank you for explanation :)